Application/src/view/visuals/Draw.java

```diff
@@ -138,7 +121,8 @@ public void addClassDiagramClass (String name){
      * @param device
      */
     public void addDeploymentDiagramClass (String device){
-        allDeploymentClasses.add(new DeploymentDiagramClass(device));
+        if (!allClassDiagramClasses.contains(device))
+            allDeploymentClasses.add(new DeploymentDiagramClass(device));
```

Comment:
wait, is this hardcoded?

Reply:
no? This was added in 57, which is still open.